docs/unit-testing-a-new-schema.md

```diff
@@ -0,0 +1,69 @@
+# Unit testing a new schema
+
+## Overview
+As the chs-delta-api uses a 3rd party library (See `/docs` directory for documentation on the kin-openAPI3 library) to 
+validate its requests, there is no way to directly unit test the code. This document will guide you through creating 
+unit tests which assert that your openAPI schema is working correctly.
+
+## 1. Where to add your unit tests
+Move into the `/validation/schema_testing` directory and create a new directory to contain your unit tests. Inside of your
+new directory, create a set `/request_bodies` and `/response_bodies` directories. These will contain your sample request 
+and response bodies used to unit test the schema.
+
+Finally, create a new Go file ending in `_test` (e.g. `exampleDeltaSchema_test.go`). This file will contain the unit 
+tests used to test the schema.
+
+## 2. Unit test structure
+We elected to structure the openAPI schema unit tests using a `Karate API Scenario` layout. Each unit test will have a 
+`Given`, `When`, `Then` stage and will use sample request bodies and response bodies to test the schema.
+
+All unit tests will follow the same structure:
+
+1. Load in the example request body from a text file using the `common.ReadRequestBody()` function
+2. Create a test httpRequest and set its target to your delta's endpoint, and its request's body to the previously read in
```

Comment:
I had to read this a couple of times for it to scan. Perhaps:
`2. Create and populate a test httpRequest:`
`a. Set its target to your delta's endpoint`
`b. Set its body to that loaded in step (1)`
`c. Set its headers`

The way to achieve the above in GitHub Markdown is just by indenting the subheading numbers by three spaces - I tried to quote it this way, but GitHub ate the spaces in my comment.